src/libraries/Microsoft.PowerFx.Interpreter/Marshal/TypeCoercionProvider.cs

```diff
@@ -238,16 +238,15 @@ internal static bool TryCoerceTo(this FormulaValue value, FormattingInfo formatI
         public static bool TryCoerceToRecord(this RecordValue value, RecordType targetType, out RecordValue result)
         {
             result = null;
-            int n = value.Fields.Count();
-            var recordResult = new NamedValue[n];
-
-            for (int i = 0; i < n; i++)
+            var recordResult = new List<NamedValue>();
+            
+            foreach (var sourceField in value.Fields)

```

Comment:
>foreach (var sourceField in value.Fields) [](http://example.com/codeflow?start=12&length=41)

We should directly loop over targetType,
If fields is present in `value` add it, else fill the `blank()`.
That way we avoid separate loop at line 272, And logic is easy to reason about!

Reply:
If we loop target inside source, it is o(n^2) computation, seperate loop only cost o(2*n). I don't think we want it.

Reply:
Not inside the source, I meant to say just looping over target, O(n) (instead of current O(2n))

Reply:
i see, you mean use target instead of source. make sense then.